konnectivity-socks5-proxy/main.go

```diff
@@ -223,6 +276,13 @@ func (d proxyResolver) Resolve(ctx context.Context, name string) (context.Contex
 		address, err := d.guestClusterResolver.resolve(ctx, name)
 		if err != nil {
 			l.Error(err, "failed to look up address from guest cluster")
+
+			if d.resolveFromManagementCluster {
+				l.Info("Fallback to management cluster resolution")
+				d.dnsFallback.set(true)
```

Comment:
this should go in a runOnce block, since after setting it to true once, we don't want to ever set it to true again.

Reply:
as in https://github.com/openshift/hypershift/blob/907b7d504d633a11066a8998efceb46e03a0b0ed/control-plane-operator/hostedclusterconfigoperator/controllers/resources/resources.go#L504-L510